extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -359,21 +359,18 @@ async function exists(path: string): Promise<boolean> {
  * If there is any selected option that doesn't match, return false.
  * Return true if all clauses match.
  */
-export function processWhenClause(when: string, selectedOptions: { option: string, value: string }[]): boolean {
+export function processWhenClause(when: string | undefined, selectedOptions: { option: string, value: string }[]): boolean {
 	if (when === undefined || when.toString().toLowerCase() === 'true') {
 		return true;
 	} else {
 		const expected = when.replace(/\s/g, '').split('&&').sort();
-		let actual: string[] = [];
-		selectedOptions.forEach(option => {
-			actual.push(`${option.option}=${option.value}`);
-		});
-
-		for (let i = 0; i < expected.length; i++) {
-			if (actual.length === 0 || actual.indexOf(expected[0]) === -1) {
-				return false;
+		const actual = selectedOptions.map(option => `${option.option}=${option.value}`);
+		let allSatisfied = true;
+		expected.forEach(expectedOption => {
```

Comment:
Right, forgot that this is a function so you can't return from it directly. In that case I'd suggest just using the for loop as you had previously and just have this check in each iteration. That way you short-circuit the evaluation as soon as you find an option that doesn't exist/isn't true. 

```
if (actual.indexOf(expected[0]) === -1) {	
   return false;	
}
```